### DISCUSSION TITLE
Mlflow cacheing references to methods of models loaded with load_model( )

### DISCUSSION THREAD

**Discussion by danielmarostica** [mlflow/mlflow#8905] [Q&A]
Suppose I have 2 Mlflow models, A and B.

Model A has a custom method `model.feature_engineering()` which creates the features `f1` and `f2`.

Model B also has a custom method `model.feature_engineering()` which creates a feature `f1`.

The methods are in a custom submodule called `features` and are pulled from the model_impl by the application right after loading the model.

I deploy an app with both models' artifacts in the folder `/app/models/`

I start the application an run `loaded_model = mlflow.pyfunc.load_model("/app/models/model_A")`. I send an API request. Prediction goes on smoothly. 

But when I run `loaded_model = mlflow.pyfunc.load_model("/app/models/model_B")`, and send another request, the traceback at the feature_engineering line says feature `f2` was not found in the dataframe axis (but it shouldn't look for it, as it is a feature of model A only), and points to the path of the code of model A! (`/app/models/model_A/code/features.py`).

Summing up: the first model I load after starting the application is kept under some caching mechanism. Even after loading another model in the same variable name `loaded_model` the predictions keep being made using the first loaded model after the start.

If I restart the application and load Model B first, it works.

Any ideas on why this is happening? I can try to prepare a reproducible example if no one has stumbled across it.

Versions tried:
Python 3.7 and 3.8
Mlflow 0.24.2 and 2.3.1

**Comment by danielmarostica** [comment]
Found a solution as simple as:

sys.modules.pop('features', None)`
